pandera/backends/polars/builtin_checks.py

```diff
@@ -180,42 +179,24 @@ def notin(data: PolarsData, forbidden_values: Iterable) -> pl.LazyFrame:
     )
 
 
-@register_builtin_check(
-    error="str_matches('{pattern}')",
-)
-def str_matches(
-    data: PolarsData,
-    pattern: Union[str, re.Pattern],
-) -> pl.LazyFrame:
-    """Ensure that string values match a regular expression.
-
-    :param data: NamedTuple PolarsData contains the dataframe and column name for the check. The keys
-                to access the dataframe is "dataframe" and column name using "key".
-    :param pattern: Regular expression pattern to use for matching
-    """
-
-    return data.dataframe.select(
-        pl.col(data.key).str.contains(pattern=pattern).alias(CHECK_OUTPUT_KEY)
-    )
-
-
 @register_builtin_check(
     error="str_contains('{pattern}')",
 )
 def str_contains(
     data: PolarsData,
-    pattern: str,
+    pattern: re.Pattern,
```

Comment:
This should be consistent with the public Check api: https://github.com/unionai-oss/pandera/blob/main/pandera/api/checks.py#L430
```suggestion
    pattern: Union[str, re.Pattern],
```